### PR TITLE
feat: Update migrations postgres URI parsing and add tests

### PR DESCRIPTION
**Pull request by fbis251**
This PR updates the parsing and formatting of the URI that is used when applying migrations and adds tests for sample input connStrings

As of 94d7d68539b6980cece521b428733f6c344bc268, using a connString with a non-default port number will ignore the custom port and default to `5432` in `PgBackend.initializeDB` when attempting to run migrations. The current connString formatting code will read `pgxCfg.Host` but not `pgxCfg.Port` when generating the new connString

- `postgresq://localhost:5433/neoq?sslmode=disable` - Example input connString
- `postgresq://localhost/neoq?sslmode=disable&x-migrations-table=neoq_schema_migrations` - connString that is actually used by PgBackend to try to apply migrations, note the default `5432` port number will be used since it's not specified

https://github.com/acaloiaro/neoq/blob/94d7d68539b6980cece521b428733f6c344bc268/backends/postgres/postgres_backend.go#L356-L361 

Note that pq-style URLs are not supported, please let me know if you would like support for them

### CODE REVIEW

File: backends/postgres/postgres_backend_test.go [hunk 1]
```diff
@@ -773,3 +774,176 @@ func Test_ConnectionTimeout(t *testing.T) {
...
+			wantErr: false,
+		},
+
+		// Inputs that cause errors
+		{
+			name:    "pq-style input returns error",
+			input:   "host=localhost port=5432 dbname=mydb connect_timeout=10",
+			want:    "",
+			wantErr: true,
+		},
```
Comment (fbis251) [inline backends/postgres/postgres_backend_test.go:917]: @acaloiaro pq-style strings are expected to cause an error with the current implementation in my PR. Please let me know if you'd like me to add support for them

Comment (acaloiaro) [reply backends/postgres/postgres_backend_test.go:917]: I would prefer pq-style connection strings to be supported if it's not too much trouble.

**Review by acaloiaro — COMMENTED** [review]
@fbis251 This looks great, and thank you for the contribution. 

I'd prefer not to regress support for libpq compatiable connection strings (https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING) 

But with that said, I also don't personally use them, and I don't find them to be terribly common in other projects. So I'm not a huge advocate for them, I'd just prefer not to regress support.

If we do allow support to regress, we'll want to flag the commit with `break: ...` so that the releaser adds it to the changelog as a breaking change. We'll also want to update any docs related to connection strings. I recall mentioning the two formats somewhere.

**Comment by acaloiaro** [comment]
Hi @fbis251 -- are you still interested in moving this PR forward?

File: backends/postgres/postgres_backend_test.go (hunk 1 above)
Comment (fbis251) [reply backends/postgres/postgres_backend_test.go]: hi @acaloiaro I apologize about the delay, I just pushed up commit c5d1bfcabe75cf79918f0393e14145333e00416a which adds support for pq-style strings.

The input string will still go through `pgx.ParseConfig` which will make sure it was in a valid format before it's returned unmodified.

Let me know what you think

**Review by acaloiaro — DISMISSED** [review]
This looks good!

Now that it's ready, could you squash it down to a single commit and prefix the commit message with `feat:`? 

e.g. `feat: Improve Postgres URI parsing`

This helps build the changelog.

**Comment by fbis251** [comment]
@acaloiaro I squashed, rebased against main, and updated the PR title to add the `feat:` prefix

**Comment by acaloiaro** [comment]
@fbis251 shippin' dreams, thanks again.

**Pull request merged**